charts/kubearchive/templates/service_account.yaml

```diff
@@ -19,3 +19,9 @@ kind: ServiceAccount
 metadata:
   name: {{ .Values.apiServer.name }}
   namespace: {{ .Values.kubearchive.namespace }}
+---
```

Comment:
why have another service account `kubearchive-privileged-user` to use for token reviews instead of granting the service account ` kubearchive-api-server` the role to do token reviews instead

Reply:
I think the privileged user is granted the `view` cluster, not permissions to use token reviews. Its here: https://github.com/kubearchive/kubearchive/pull/55/files/dd96f229bbe3bc0f85742f18e514c4141af9bc73#diff-b68ce0b4f2cb84ca9c9429162a3b4fd00358b3e93c215148d26494da45e9ee99R33

Reply:
you're right, `kubearchive-privileged-user` is granted the `view` role not token reviews.

Looking at the changes in development.md, is this service account just meant for testing?

Reply:
Yes, just for testing. See https://github.com/kubearchive/kubearchive/pull/55#discussion_r1632650976

Reply:
I'm solving this as the user is for testing purposes. With the new name `kubearchive-test-sa` should be more clear

Reply:
Do we really need to have `-sa` on a service account name?

I personally am not a fan.

Reply:
I initally thought about having `user` in the name, because a service account is a user. We can change it to `kubearchive-test-user`.

The `test` part over the `privileged` based on the discussions on this an other thread I think is a good replacement, as the main point is to state that is a service account for testing.

Reply:
My point is the `-sa` or -`user` is redundant.  It's a service account. If I try want to retrieve it, I have to specify that I want a service account, i.e. `kubectl get sa`.  If I run `kubectl get all`, it's listed under service accounts.

If we do it here, are we going to do it also for every role, rolebinding, deployment, etc?

I've added this to the agenda for Thursday.

Reply:
Ok let's discuss it on Thursday.

BTW, I would usually say that we shouldn't specify what the resource is in the naming (the same way it isn't encourage to have the type included in a variable).

But, I found this a special case. The user is going to retrieve a token from a service account when doing a `curl` like the one described in the `DEVELOPMENT.md` file:

```
curl --cacert ca.crt localhost:8081/apis/apps/v1/deployments \
    -H "Authorization: Bearer $(kubectl create token kubearchive-test-sa)" \
   localhost:8081/apis/apps/v1/deployments
```

I think that having `kubearchive-test` is less clear. So I would normally go with your suggestion but treat this as an exception to the rule.

I think we aren't including the type of the resource in the rest of the helm chart values, this is the only one.

Reply:
It's on the agenda for tomorrow. But I still don't find your argument convincing. In your `curl` command is a nested `kubectl` command.  That command requires a service account. From the help:
```
[gallen@gallen-thinkpadp1gen5 kubearchive] (merge-operator-build)$ kubectl create token --help
Request a service account token.
...
[gallen@gallen-thinkpadp1gen5 kubearchive]
```